snakefmt/snakefmt.py

```diff
@@ -331,15 +333,23 @@ def main(
                         out_handle.write(formatted_content)
 
     if check:
-        if files_with_errors > 0:
-            logging.info(f"{files_with_errors} file(s) contains errors 🤕")
-            ctx.exit(CheckExitCode.ERROR.value)
+        if files_unchanged == len(files_to_format):
+            logging.info(
+                f"All {len(files_to_format)} file(s) would be left unchanged 🎉"
+            )
+            ctx.exit(ExitCode.NO_CHANGE.value)
+        elif files_with_errors > 0:
+            exit_value = ExitCode.ERROR.value
         elif files_changed > 0:
+            exit_value = ExitCode.WOULD_CHANGE.value
+
+        if files_with_errors > 0:
+            logging.info(f"{files_with_errors} file(s) raised parsing errors 🤕")
```

Comment:
This could just be inserted below line 342 and this `if` statement removed as it checks the same thing?

Reply:
I don't think so. I made this change because previously, if for eg we had 5 files with errors and 5 that would be change, we only output that 5 would be changed. I prefer having both pieces of information. So there's an exlusive if/elif setting the exit code (based on a priority: error code > changes_code) but then a global reporting.

Reply:
No problem :+1: